File: src/main/java/me/nylestroke/nylemod/fluid/ModFluids.java
package me.nylestroke.nylemod.fluid;

import me.nylestroke.nylemod.NylemodExample;
import net.minecraft.fluid.FlowableFluid;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class ModFluids {
    public static final FlowableFluid HONEY_STILL = register("honey_still", new HoneyFluid.Still());
    public static final FlowableFluid HONEY_FLOWING = register("honey_flowing", new HoneyFluid.Flowing());

    private static FlowableFluid register(String name, FlowableFluid flowableFluid) {
        return Registry.register(Registry.FLUID, new Identifier(NylemodExample.MOD_ID, name), flowableFluid);
    }
}

File: src/main/kotlin/me/nylestroke/nylemod/fluid/ModFluids.kt
package me.nylestroke.nylemod.fluid

import me.nylestroke.nylemod.NylemodExample
import net.minecraft.fluid.FlowableFluid
import net.minecraft.fluid.Fluid
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry

object ModFluids {
    val HONEY_STILL: FlowableFluid = register("honey_still", HoneyFluid.Still())
    val HONEY_FLOWING: FlowableFluid = register("honey_flowing", HoneyFluid.Flowing())
    private fun register(name: String, flowableFluid: FlowableFluid): FlowableFluid {
        return Registry.register<Fluid, FlowableFluid>(Registry.FLUID, Identifier(NylemodExample.MOD_ID, name), flowableFluid)
    }
}